presto-main/src/main/java/com/facebook/presto/cost/StatsNormalizer.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.spi.type.BigintType;
+import com.facebook.presto.spi.type.BooleanType;
+import com.facebook.presto.spi.type.DateType;
+import com.facebook.presto.spi.type.DecimalType;
+import com.facebook.presto.spi.type.IntegerType;
+import com.facebook.presto.spi.type.SmallintType;
+import com.facebook.presto.spi.type.TinyintType;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.Symbol;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static com.facebook.presto.cost.SymbolStatsEstimate.UNKNOWN_STATS;
+import static com.facebook.presto.cost.SymbolStatsEstimate.ZERO_STATS;
+import static java.lang.Double.NaN;
+import static java.lang.Double.isNaN;
+import static java.lang.Math.floor;
+import static java.lang.Math.pow;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Makes stats consistent
+ */
+public class StatsNormalizer
+{
+    public PlanNodeStatsEstimate normalize(PlanNodeStatsEstimate stats, Map<Symbol, Type> types)
+    {
+        return normalize(stats, Optional.empty(), types);
+    }
+
+    public PlanNodeStatsEstimate normalize(PlanNodeStatsEstimate stats, Collection<Symbol> outputSymbols, Map<Symbol, Type> types)
+    {
+        return normalize(stats, Optional.of(outputSymbols), types);
+    }
+
+    private PlanNodeStatsEstimate normalize(PlanNodeStatsEstimate stats, Optional<Collection<Symbol>> outputSymbols, Map<Symbol, Type> types)
+    {
+        PlanNodeStatsEstimate.Builder normalized = PlanNodeStatsEstimate.buildFrom(stats);
+
+        Predicate<Symbol> symbolFilter = outputSymbols
```

Comment:
`isSymbolInOutput`?

Reply:
I am hesitant. Sometimes output symbols are not passed here (`Optional.empty`), so we don't know what are the output symbols.